src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/UrlSyndicationContent.cs

```diff
@@ -3,55 +3,35 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Xml;
-using System.Runtime.CompilerServices;
 
 namespace System.ServiceModel.Syndication
 {
     // NOTE: This class implements Clone so if you add any members, please update the copy ctor
     public class UrlSyndicationContent : SyndicationContent
     {
         private string _mediaType;
-        private Uri _url;
 
         public UrlSyndicationContent(Uri url, string mediaType) : base()
         {
-            if (url == null)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("url");
-            }
-            _url = url;
+            Url = url ?? throw new ArgumentNullException(nameof(url));
             _mediaType = mediaType;
         }
 
-        protected UrlSyndicationContent(UrlSyndicationContent source)
-            : base(source)
+        protected UrlSyndicationContent(UrlSyndicationContent source) : base(source)
         {
-            if (source == null)
-            {
```

Comment:
I know base(source) does a null check, but it does it via SyndicationContent.CopyAttributeExtensions. That's too far removed such that if someone were to modify that method and change the null check, it wouldn't be obvious that this code relies on it. Please restore the null check as we shouldn't rely on side-effects to do our input validation.

Reply:
We now have tests that would verify this behaviour - I don't see any need to keep these checks.

Reply:
Tests can be refactored/removed. This is still relying on external code to null check and relying on external code to test that the null check is still there. Could you put a Debug.Assert null check here instead so that we're not relying on the existence of external code to catch this.